.changeset/new-dolls-dress.md

```diff
@@ -0,0 +1,5 @@
+---
```

Comment:
Another note to myself that while there are currently changesets in this diff, they apply to earlier PRs in the stack. I'm kind of on the fence about whether this change warrants one: it's _almost entirely_ focused on engine internals. I think the only client-facing change is one which we'd like to eliminate from the client interface entirely (i.e. changes to `RepeatRangeNode.definition` and `RepeatInstanceNode.definition`).